### PR TITLE
cmd/govim: add support for "fill struct" code action

### PR DESCRIPTION
**Pull request by leitzler**
Adds a new command, "GOVIMFillStruct", that populate fields with zero
value in a struct using gopls new code action "Fill struct".

Currently it must be called on an empty struct with the cursor above the
right bracket '}', see golang/go#39506.

### CODE REVIEW

File: cmd/govim/fillstruct.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			buri := b.URI()
+			if euri != buri {
+				return fmt.Errorf("got edits for file %v, but buffer is %v", euri, buri)
+			}
+			if ev := int(math.Round(dc.TextDocument.Version)); ev > 0 && ev != b.Version {
+				return fmt.Errorf("got edits for version %v, but current buffer version is %v", ev, b.Version)
+			}
+			edits := dc.Edits
+			if len(edits) != 0 {
+				if err := v.applyProtocolTextEdits(b, edits); err != nil {
```
Comment (myitcv) [inline cmd/govim/fillstruct.go:46]: How many calls to `applyProtocolTextEdits` are you expecting here? I'm assuming the answer is one.

It might be slightly nicer to assert that and drop the `for` loops. Because otherwise it gives the impression at first reading that we're going to be calling `applyProtocolTextEdits` multiple times... which would be crazy because we'd want to gather all the edits first and then call once.

Comment (leitzler) [reply cmd/govim/fillstruct.go:52]: Indeed, good idea, I updated it according to your suggestion!

It originated from the fact that we do get an array back from gopls. And since the feature is new and going to get a overview soon I went for the safe thing. If we (for some reason) get more than one edit we will end up throwing an error and rely on users reporting it as an issue (since our test probably won't cover an edge case).
